app/assets/javascripts/directives/containers/chart-data-mixin.js

```diff
@@ -0,0 +1,102 @@
+angular.module('containers').factory('ChartsDataMixin', ['$timeout', '$q', function chartDataMixinFactory ($timeout, $q) {
+
+    var sparklineTimeTooltip = function(d) {
+        return '  <table class="c3-tooltip">' +
+            '    <tbody>' +
+            '      <td class="value">' +  d[0].x.toLocaleTimeString() + '</td>' +
+            '      <td class="value text-nowrap">' +  d[0].value + ' ' + d[0].name + '</td>' +
+            '    </tbody>' +
+            '  </table>';
```

Comment:
If we can't, can we at least avoid treating some things that don't sound like HTML, as trusted HTML content? I don't know enough about what's going on here to claim we *currently* can't trust it... but unless it's actually intended to be HTML, this is at least how accidents happen later.